nanome_vault/VaultServer.py

```diff
@@ -59,6 +59,36 @@ def _parse_path(self):
         except:
             Logs.error('Error parsing path:\n', traceback.format_exc())
 
+    def _is_auth_valid(self, path):
+        if not VaultServer.enable_auth:
+            return True
+
+        auth = self.headers.get('authorization')
+        if auth is None:
+            return False
+
+        token = auth.split(' ')[-1]
+        cached = VaultServer.auth_cache.get(token, None)
+
+        if cached is None:
+            headers = {'authorization': auth}
+            r = requests.get('https://api.nanome.ai/user/session', headers=headers)
```

Comment:
_is_auth_valid is executing on a thread, right? Just checking to make sure requests.get doesn't freeze the web server in case of timeout

Reply:
Yes, the `HTTPServer` is using `ThreadingMixIn` which spawns a new thread for each request.